### ISSUE TITLE
Setup using use-package

### ISSUE THREAD

**Issue by tpadioleau** [d12frosted/flyspell-correct#75]
Hi,

I tried to use `flyspell-correct-helm` flavor using the setup you propose
```elisp
(use-package flyspell-correct-helm
  :bind ("C-;" . flyspell-correct-wrapper)
  :init
  (setq flyspell-correct-interface #'flyspell-correct-helm))
```
It doesn't seem to work for me. The key binding is not set, it is actually still bound to `flyspell-auto-correct-previous-word`. Moreover if I use manually the `flyspell-correct-wrapper` function, it complains saying `helm-M-x-execute-command: Symbol’s function definition is void: flyspell-correct-helm`.

The simpler setup you suggest
```elisp
(require 'flyspell-correct-helm)
(define-key flyspell-mode-map (kbd "C-;") 'flyspell-correct-wrapper)
```
is working for me (key binding and it uses helm version). But I notice that it doesn't seem to do operations in the same order as in the use-package setup.

I think there are two problems with the use-package setup:
- The key binding is hidden by `flyspell-mode-map`, a first fix could be
```elisp
(use-package flyspell-correct-helm
  :after flyspell ;; to make sure flyspell-mode-map is defined
  :bind (:map flyspell-mode-map ("C-;" . flyspell-correct-wrapper))
  :init
  (setq flyspell-correct-interface #'flyspell-correct-helm))
```
it makes it closer to the simple setup you suggest.
- Even if the key binding works, the package is not loaded, maybe because `flyspell-correct-wrapper` is not part of `flyspell-correct-helm` ? A workaround could be to split into two distinct use-package declarations
```elisp
(use-package flyspell-correct
  :after flyspell
  :bind (:map flyspell-mode-map ("C-;" . flyspell-correct-wrapper)))

(use-package flyspell-correct-helm
  :after flyspell-correct)
```
This way `flyspell-correct-helm` is loaded as soon as `flyspell-correct` is, for example when `C-;` is pressed. `(setq flyspell-correct-interface #'flyspell-correct-helm)` doesn't seem needed because it is already in `flyspell-correct-helm.el`. Finally if one doesn't want to use a key-binding, it can be replaced by explicitly deferring with `:defer t`.

Thanks

**Comment by d12frosted** [comment]
Hey @tpadioleau 

Thank you very much for your feedback and for time you spent digging into it! In my own configurations I am also split `flyspell-correct` and `flyspell-correct-<interface>` configurations, because it makes sense to.

I would gladly accept PR improving documentation.

**Comment by d12frosted** [comment]
Fixed via #76
